arangod/Cluster/ClusterInfo.cpp

```diff
@@ -896,66 +898,80 @@ void ClusterInfo::loadPlan() {
-            newCollection = std::make_shared<LogicalCollection>(  // create collection
-                *vocbase, collectionSlice, true, newPlanVersion  // args
-            );
-          }
 
-          auto& collectionName = newCollection->name();
-
-          bool isBuilding = isCoordinator &&
-                            arangodb::basics::VelocyPackHelper::getBooleanValue(
-                                collectionSlice, StaticStrings::AttrIsBuilding, false);
-          if (isCoordinator) {
-            // copying over index estimates from the old version of the
-            // collection into the new one
-            LOG_TOPIC("7a884", TRACE, Logger::CLUSTER)
-                << "copying index estimates";
-
-            // it is effectively safe to access _plannedCollections in
-            // read-only mode here, as the only places that modify
-            // _plannedCollections are the shutdown and this function
-            // itself, which is protected by a mutex
-            auto it = _plannedCollections.find(databaseName);
-            if (it != _plannedCollections.end()) {
-              auto it2 = (*it).second.find(collectionId);
-
-              if (it2 != (*it).second.end()) {
-                try {
-                  auto estimates = (*it2).second->clusterIndexEstimates(false);
-
-                  if (!estimates.empty()) {
-                    // already have an estimate... now copy it over
-                    newCollection->setClusterIndexEstimates(std::move(estimates));
+            if (isCoordinator) {
+              // copying over index estimates from the old version of the
+              // collection into the new one
+              LOG_TOPIC("7a884", TRACE, Logger::CLUSTER)
+                  << "copying index estimates";
+
+              // it is effectively safe to access _plannedCollections in
+              // read-only mode here, as the only places that modify
+              // _plannedCollections are the shutdown and this function
+              // itself, which is protected by a mutex
+              auto it = _plannedCollections.find(databaseName);
+              if (it != _plannedCollections.end()) {
+                auto it2 = (*it).second.find(collectionId);
+
+                if (it2 != (*it).second.end()) {
+                  try {
+                    auto estimates = (*it2).second.first->clusterIndexEstimates(false);
+
+                    if (!estimates.empty()) {
+                      // already have an estimate... now copy it over
+                      newCollection->setClusterIndexEstimates(std::move(estimates));
+                    }
+                  } catch (...) {
+                    // this may fail during unit tests, when mocks are used
                   }
-                } catch (...) {
-                  // this may fail during unit tests, when mocks are used
                 }
               }
             }
+          } else {
+            LOG_TOPIC("52632", TRACE, Logger::CLUSTER)
```

Comment:
In the case of recycling an existing LogicalCollection object, should we update the plan version in the collection object?
I saw above that `LogicalCollection` gets the planVersion as one of its constructor parameters. Not sure what this is used for, but that version will remain static now in case a LogicalCollection object is recycled.

Reply:
Good point. I do not know for what it is used, but I will update the plan version.